### ISSUE TITLE
Filtering not working while relationTo is array of collections 

### ISSUE THREAD

**Issue by geminigeek** [payloadcms/payload#3958]
### Link to reproduction

na

### Describe the Bug

i have a simple comment collection in basic payload create app which is having a relation with posts and blurbs collection, blurb collection is same as post 

comment collection fields are 
```
fields: [
    {
      name: 'title',
      type: 'text',
    },
   
    {
      name: "contentItem",
      type: "relationship",
      relationTo: ["posts", "blurbs"],

    },

    {
      name: 'status',
      type: 'select',
      options: [
        {
          value: 'moderation',
          label: 'In Moderation',
        },
        {
          value: 'published',
          label: 'Published',
        },
      ],
      defaultValue: 'moderation',
      admin: {
        position: 'sidebar',
      }
    }
  ],
```

i am trying to filter records  where comment status is moderation, for a post item , while the selection can be made in ui the results are not according to filter in ui the query string as captured in chrome devtool is 

```
depth: 0
draft: true
limit: 10
page: 1
where[or][0][and][0][contentItem][equals][relationTo]: posts
where[or][0][and][0][contentItem][equals][value]: 654215843bb61a5bc9bb3593
where[or][0][and][1][status][equals]: moderation
invoke: 09b9bb2a-1781-4a6e-9216-a932dbf0aa65
```
here is an image of ui 

![image](https://github.com/payloadcms/payload/assets/3751322/817e26b2-a789-4663-8dec-6fd73efa1056)

### To Reproduce

create basic blog payload project , copy post collection rename to blurbs, add comment collection ,

```
import { CollectionConfig } from 'payload/types';

const Comments: CollectionConfig = {
  slug: 'comments',
  admin: {
    defaultColumns: ['title', 'author', 'category', 'tags', 'status'],
    useAsTitle: 'title',
  },
  access: {
    read: () => true,
  },
  fields: [
    {
      name: 'title',
      type: 'text',
    },

    {
      name: "contentItem",
      type: "relationship",
      relationTo: ["posts", "blurbs"],

    },

    {
      name: 'status',
      type: 'select',
      options: [
        {
          value: 'moderation',
          label: 'In Moderation',
        },
        {
          value: 'published',
          label: 'Published',
        },
      ],
      defaultValue: 'moderation',
      admin: {
        position: 'sidebar',
      }
    }
  ],
}

export default Comments;
```

create a post 
create a blurb

create a comment on post
create a comment on blurb 

try to filter comment for post or blurb its not working 

### Payload Version

2.0.14

### Adapters and Plugins

db-mongodb,bundler-webpack

**Comment by JessChowdhury** [comment]
Hi @geminigeek - please see my reply on this similar [issue](https://github.com/payloadcms/payload/issues/3761#issuecomment-1782942285).

TLDR; this is expected behavior. Polymorphic relationships return a different data shape which is not supported by the collection filters. 

I've added some clarification to our [Relationship Field](https://payloadcms.com/docs/fields/relationship) documentation, which will be live when [this PR](https://github.com/payloadcms/payload/pull/3976) is merged.

It may be possible to add support for this in the future when a better solution is available. 

Let me know if you have any questions about this or need further help. Thanks!

**Comment by geminigeek** [comment]
hi jess,

i can pull the intended data from the local api , why cant we run same query , if we have field.relationTo and field.value, 
```
depth: 0
draft: true
limit: 10
page: 1
where[or][0][and][0][contentItem][equals][relationTo]: posts
where[or][0][and][0][contentItem][equals][value]: 654215843bb61a5bc9bb3593
where[or][0][and][1][status][equals]: moderation
invoke: 09b9bb2a-1781-4a6e-9216-a932dbf0aa65
```
i can see we have an upcoming  flag/header in a pr https://github.com/payloadcms/payload/pull/3324 that identify if the request is coming from admin ui, the same flag can be used to change querying for these kind of filters.

pulling data from local api
```
const result= await req.payload.find({
          collection: "comments",
          where: {
            and: [{
              "contentItem.relationTo": { equals: "posts"},
              "contentItem.value": { equals: "654215843bb61a5bc9bb3593"},
              "status": { equals: "moderation"},

            }]
          },
          depth: 0

        })
```
